internal/hooks/v0hooks/manager.go

```diff
@@ -31,202 +31,84 @@ const (
-			} else {
-				return nil, internalServerError("Failed to trigger auth hook, error making HTTP request").WithInternalError(err)
-			}
-		}
+}
 
-		defer rsp.Body.Close()
+func (o *Manager) InvokeHook(
+	conn *storage.Connection,
+	r *http.Request,
+	input, output any,
+) error {
+	return o.invokeHook(conn, r, input, output)
+}
 
-		switch rsp.StatusCode {
-		case http.StatusOK, http.StatusNoContent, http.StatusAccepted:
-			// Header.Get is case insensitive
-			contentType := rsp.Header.Get("Content-Type")
-			if contentType == "" {
-				return nil, badRequestError(apierrors.ErrorCodeHookPayloadInvalidContentType, "Invalid Content-Type: Missing Content-Type header")
-			}
-			mediaType, _, err := mime.ParseMediaType(contentType)
-			if err != nil {
-				return nil, badRequestError(apierrors.ErrorCodeHookPayloadInvalidContentType, fmt.Sprintf("Invalid Content-Type header: %s", err.Error()))
-			}
-			if mediaType != "application/json" {
-				return nil, badRequestError(apierrors.ErrorCodeHookPayloadInvalidContentType, "Invalid JSON response. Received content-type: "+contentType)
-			}
-			if rsp.Body == nil {
-				return nil, nil
-			}
-			limitedReader := io.LimitedReader{R: rsp.Body, N: PayloadLimit}
-			body, err := io.ReadAll(&limitedReader)
-			if err != nil {
-				return nil, err
-			}
-			if limitedReader.N <= 0 {
-				// check if the response body still has excess bytes to be read
-				if n, _ := rsp.Body.Read(make([]byte, 1)); n > 0 {
-					return nil, unprocessableEntityError(apierrors.ErrorCodeHookPayloadOverSizeLimit, fmt.Sprintf("Payload size exceeded size limit of %d bytes", PayloadLimit))
-				}
-			}
-			return body, nil
-		case http.StatusTooManyRequests, http.StatusServiceUnavailable:
-			retryAfterHeader := rsp.Header.Get("retry-after")
-			// Check for truthy values to allow for flexibility to switch to time duration
-			if retryAfterHeader != "" {
-				continue
-			}
-			return nil, internalServerError("Service currently unavailable due to hook")
-		case http.StatusBadRequest:
-			return nil, internalServerError("Invalid payload sent to hook")
-		case http.StatusUnauthorized:
-			return nil, internalServerError("Hook requires authorization token")
-		default:
-			return nil, internalServerError("Unexpected status code returned from hook: %d", rsp.StatusCode)
-		}
-	}
-	return nil, nil
+func (o *Manager) RunHTTPHook(
+	r *http.Request,
+	hookConfig conf.ExtensibilityPointConfiguration,
+	input any,
+) ([]byte, error) {
+	return o.runHTTPHook(r, hookConfig, input)
 }
 
-// invokePostgresHook invokes the hook code. conn can be nil, in which case a new
+// invokeHook invokes the hook code. conn can be nil, in which case a new
 // transaction is opened. If calling invokeHook within a transaction, always
 // pass the current transaction, as pool-exhaustion deadlocks are very easy to
 // trigger.
-func (a *API) invokeHook(conn *storage.Connection, r *http.Request, input, output any) error {
+func (o *Manager) invokeHook(
```

Comment:
nit: would `m` be a better name for the struct here? `o` makes me think of output for some reason

Reply:
I often use "o" for method receivers much like you use "i" for indices in for loops. I understand projects often like custom letters for each struct, but it has the downside that you can't easily tell when you're pointing to the methods receiver. I continue to use the same receiver for existing structs, but would prefer using "o" for new ones.